src/txmempool.h

```diff
@@ -485,7 +485,13 @@ class CTxMemPool
         >
     > indexed_transaction_set;
 
-    mutable CCriticalSection cs;
+    /**
+     * Mutex that guards reads.
```

Comment:
`Recursive mutex ...`

Reply:
Recursive is already clear from the variable name and I plan to make it non-recursive